localstack-core/localstack/aws/protocol/parser.py

```diff
@@ -1096,28 +1096,32 @@ def _get_serialized_name(self, shape: Shape, default_name: str, node: dict) -> s
         For example, both works for the TagQueue operation:
         - Using the proper serialized name "Tag": Tag.1.Key=key&Tag.1.Value=value
         - Using the member name "Tag" in the parent structure: Tags.1.Key=key&Tags.1.Value=value
+        - Using "Name" to represent the Key for a nested dict: MessageAttributes.1.Name=key&MessageAttributes.1.Value.StringValue=value
+            resulting in {MessageAttributes: {key : {StringValue: value}}}
         The Java SDK implements the second variant: https://github.com/aws/aws-sdk-java-v2/issues/2524
         This has been approved to be a bug and against the spec, but since the client has a lot of users, and AWS SQS
         supports both, we need to handle it here.
```

Comment:
@thrau The comment you left here was absolute gold, and made understanding the current issue and fix so much easier! :bomb: :fire: :rocket:

Reply:
i could spot alex' handwriting, but had to double check as well :smile: